totaladapters/src/main/java/pl/arturborowy/adapters/common/BasicAdapterContract.kt

```diff
@@ -0,0 +1,30 @@
+package pl.arturborowy.adapters.common
+
+import android.view.View
+import android.view.ViewGroup
+import org.koin.core.KoinComponent
+import pl.arturborowy.adapters.HasLayoutResId
+import pl.arturborowy.adapters.ViewStylist
+import pl.arturborowy.util.ViewInflater
+
+internal interface BasicAdapterContract<ItemT, ViewT : View> :
+        HasLayoutResId, ViewStylist<ViewT, ItemT>, KoinComponent, UpdateItemsAdapter<ItemT> {
+
+    var items: Collection<ItemT>
+
+    val notifyDataSetChanged: () -> Unit
+
+    override fun updateItems(items: Collection<ItemT>) {
+        this.items = items
+        notifyDataSetChanged()
+    }
+
+    fun createNewView(viewInflater: ViewInflater, parent: ViewGroup) =
+            viewInflater.inflate(getLayoutResId(), parent) as ViewT
```

Comment:
Maybe we should check the type and inform the user that top view from `getLayoutResId()` has to conform with ViewT? An exception with helpful message would be nice.

Reply:
Done at feature/ab_basic_adapter_contract_tess